libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/EntitySection.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.Builder.Parsers.LU.Parser
+{
+    /// <summary>SectionEntity class.</summary>
+    public partial class SectionEntity
+    {
+        // TODO: pass this constant to a helper class.
+        private readonly char[] _invalidCharsInIntentOrEntityName = { '<', '>', '*', '%', '&', ':', '\\', '$' };
+
+        public SectionEntity(LUFileParser.EntitySectionContext parseTree)
+        {
```

Comment:
This is a public method, pls null check `parseTree`

Reply:
If other validations or pre-conditions are required around the state of ParseTree, we should validate them as well.

Reply:
Solved